pkg/webhooks/jobset_webhook.go

```diff
@@ -309,9 +309,9 @@ func (j *jobSetWebhook) ValidateUpdate(ctx context.Context, old, newObj runtime.
 	}
 	mungedSpec := js.Spec.DeepCopy()
 
-	// Allow pod template to be mutated for suspended JobSets.
+	// Allow pod template to be mutated for suspended JobSets, or JobSets getting suspended.
 	// This is needed for integration with Kueue/DWS.
-	if ptr.Deref(oldJS.Spec.Suspend, false) {
+	if ptr.Deref(oldJS.Spec.Suspend, false) || ptr.Deref(js.Spec.Suspend, false) {
```

Comment:
```suggestion
	if ptr.Deref(oldJS.Spec.Suspend, false) || (ptr.Deref(js.Spec.Suspend, false) && js.Spec.ReplicatedJobs[*].status.startTime == nil) {
```

Wouldn't it be helpful to validate if the jobs don't have startTime?
Because if any jobs have startTime, this operation should fail due to batch/job validation.
But, I'm not sure if we should add validations deeply depending on the batch/job specifications.

@danielvegamyhre @mimowo WDYT?

Reply:
I think the flow is that we first suspend the parent JobSet, then it suspends the Jobs. So, if we added this condition, the `startTime` on running child jobs would prevent the update.

I think I should have added an e2e test in JobSet to verify the suspending works.

Reply:
Yea I’m thinking integration and e2e test solidifying the requirements that JobSet needs for working with Kueue will be a great idea.

We obviously don’t want to bring in Kueue as a dependency so I think just verifying that the updates/patches work in integration/e2e will be welcome

Reply:
> I think the flow is that we first suspend the parent JobSet, then it suspends the Jobs. So, if we added this condition, the startTime on running child jobs would prevent the update.

Uhm, that makes sense. 
But, I guess that users fall into the rabbit hole when they accidentally modify the scheduling directives against the ReplicatedJob with startTime because the jobset-controller just output errors in the controller logs, this is not directly user feedback.

Maybe we can improve these implied specifications for the JobSet users once we introduce https://github.com/kubernetes/kubernetes/issues/113221.
 
So, I'm ok with the current implementation for now.

Reply:
Actually, the e2e test would not work, because there is a deeper issue - JobSet does not delete the Jobs once suspended: https://github.com/kubernetes-sigs/jobset/issues/535. I'm trying to implement it by deleting the Jobs on suspend, and it looks promising. There are some integration tests for startupPolicy to adjust I yet need to look into.

Reply:
Let me know if you have some more context knowledge if traps I may encounter with this approach, but essentially this is the only I see, and it would mimic the Job behavior, where suspended Jobs deletes the pods.